pkg/utils/tasks/tasks.go

```diff
@@ -71,24 +121,120 @@ func (tm *TaskManager) IsStopped() bool {
 	return tm.stopped
 }
 
+// Queue a task defined by the handler to the specified channel. If the task returns an
+// error, this method queues the task for retry by sending it to the retry channel.
+func (tm *TaskManager) queueTask(ctx context.Context, handler *TaskHandler, dest chan<- *TaskHandler) {
+	// Wrap the task with retry logic
+	retry := func(ctx context.Context) (err error) {
+		if err = handler.task.Do(ctx); err == nil {
+			return nil
+		}
+
+		handler.attempts++
+		if handler.attempts < handler.conf.retries {
+			// Send the retry task to the scheduler
+			handler.retry = time.Now().Add(handler.conf.backoff.NextBackOff())
+			tm.queueTask(ctx, handler, tm.retry)
+			return nil
+		}
+
+		// Log the error so we know that all the retries failed
+		if handler.conf.err != nil {
+			log.Error().Err(err).Int("attempts", handler.attempts).Msg("task failed after retries")
+
+			if handler.conf.ctx != nil {
+				// TODO: is this a thread-safe way to create a hub for capturing exceptions?
+				hub := sentrygin.GetHubFromContext(handler.conf.ctx).Clone()
```

Comment:
I think this is a good idea, but there are a couple of things that we'll need to do to make it work:

1. The request context is going to go away in a different thread - so we need to clone before the task is queued.
2. We need a task context so the outer context can get passed in; right now no context will have the hub because it's being newly created and canceled inside of each task function. 

I'll go ahead and make that change to see if it fixes it.